README.md
Exodus
======


pixel/changelog.txt
Pixel This Change Log

10/24/14 - v1.2.1
Fixed an issue with most subreddits not working. The issue seemed to be rooted in the fact that not all subreddits
support sorting by "top", but all that I've tested seem to support sorting by time, so I switched to that.

10/24/14 - v1.2.0
Bombs got a huge boost and as a result had the reload time boost granted by an upgrade quartered
Added a prompt on a new session about the party pixel flashing, option is still in the menu
Added new upgrades for party pixel
Popping a party pixel now gives bonus pixels, purely based on your bank with no cap
Added a search term box which lets you specify either search terms or a subreddit
- Subreddits must be in the form of /r/subreddit
- Search terms can follow the q value rules at https://api.imgur.com/endpoints/gallery#gallery-search
- So you can put "kitties ext: gif" in the search box (without the quotes) and get only kitty gifs. :-)
- I have not had good luck using the boolean operators, but they should work in theory (e.g. "kitties OR puppies ext: gif")
- If you have exhausted 10 pages of results (600 images) based on the filters, it will drop you back to random gallery
- For example, if you go to a certain subreddit where girls are wild, without NSFW toggled, there is a very high chance you'll just wind up back in random
Added a NSFW Toggle box. If this is toggled, it means NSFW images are allowed. It DOES NOT mean only NSFW images will show up

9/16/14 - v1.1.4
Added night mode option (in menu tab)
Added option to stop flashing during party (in menu tab)
Added Hard Reset button (in menu tab)
Added Bomb timer (thanks http://www.reddit.com/user/Ballpit_Inspector)
Reduced time before auto next image triggers from 5 min to 30 sec
Fixed a bug causing massive lag if part of a large manual cursor was touching "above" the image\
Fixed the same image showing up multiple times
Added some more stats

9/16/14 - v1.1.3
Added version identifier in menu
Fixed stupid achievement bug which killed next image button sometimes

9/15/14 - v1.1.2
Fixed history bugs

9/15/14 - v1.1.1
Added option for number formatting
Fixed/moved lag in Opera

9/15/14 - v1.1.0
Initial Live Release
Greatly improved starting speed
Moved some necessary upgrades out from under info/stat upgrades
Reduced the price on some upgrades
Added Pixel Splitters

9/13/14 - v1.0.4
Hopefully fixed bug with loading saved image

9/12/14 - v1.0.3
Fixed a bug with missing a party pixel
Rearranged the achievements so hopefully it's less consuming

9/11/14 - v1.0.2
Fixed a bug with tracking bombs

9/11/14 - v1.0.1
Fixed a missing feature to the PaRtY PiXeL

9/11/14 - v1.0.0
Internal Release


js/Planet.js
var colors = new Array("Red","Blue","Yellow","Green","Purple","Black","White","Lime","Magenta","Cyan","Orange","Navy");
var colorHex1 = new Array("#AA0000","#0000AA","#FFFF00","#00DD00","#DD00FF","#111111","#DDDDDD","#88CC00","#FF55BB","#00FFDD","#FF7700","#111177");
var colorHex2 = new Array("#DD0000","#0000DD","#FFFF00","#00DD00","#DD00FF","#111111","#DDDDDD","#88CC00","#FF55BB","#00FFDD","#FF7700","#111177");

function PointOnCircle(radius, angle, originX, originY) {
	// Convert from degrees to radians via multiplication by PI/180        
	var x = radius * Math.cos(angle * Math.PI / 180.0) + originX;
	var y = radius * Math.sin(angle * Math.PI / 180.0) + originY;

	return new Array(x, y);
}

function Planet(starting) {
	Math.seedrandom();
	this.mods = {};
	this.mods.food = 1.0;
	this.mods.wood = 1.0;
	this.mods.metal = 1.0;
	this.cosmetic = {};
	this.cosmetic.color1ndx = Math.floor(Random(0, colors.length));
	this.cosmetic.color1 = colors[this.cosmetic.color1ndx];
	this.cosmetic.color2ndx = Math.floor(Random(0, colors.length));
	this.cosmetic.color2 = colors[this.cosmetic.color2ndx];
	
	this.radius = 400;
	this.originX = 600;
	this.originY = 400;
	this.shape1 = new Array(200, 220, 240, 190);
	this.shape2 = new Array(140, 90, 100, 200);
	
	this.satellites = new Array();
	var numSats = 2;
	if(!starting) {
		numSats = Math.floor(Math.random()*5);
	}
	for(var i = 0; i != numSats; i++) {
		this.satellites.push("Satellite "+i);
	}
	this.name = "";
	if(!starting) {
	}
}

Planet.prototype = {
	constructor: Planet,
	Draw:function(canvas) {
		var c=document.getElementById(canvas);
		var ctx=c.getContext("2d");
		ctx.globalCompositeOperation = 'source-over';
		ctx.clearRect(0, 0, 600, 800);
		ctx.fillStyle=colorHex2[this.cosmetic.color2ndx];
		ctx.globalAlpha = 0.7;
		ctx.beginPath();
		ctx.arc(this.originX,this.originY,this.radius,Math.PI*0.5,Math.PI*1.5);
		ctx.fill();
		
		ctx.globalCompositeOperation = 'source-atop';
		ctx.fillStyle=colorHex1[this.cosmetic.color1ndx];
		ctx.beginPath();
		//Planet at equ is (200,400)
		//(600,0) at north pole
		//(600,800) at south pole
		ctx.moveTo(250,475);
		ctx.bezierCurveTo(220,550,380,600,405,510);
		ctx.quadraticCurveTo(415,500,480,510);
		ctx.bezierCurveTo(520,515,530,360,500,340);
		ctx.quadraticCurveTo(475,320,510,215);
		ctx.quadraticCurveTo(525,175,470,200);
		ctx.bezierCurveTo(450,215,320,110,370,70);
		ctx.bezierCurveTo(100,100,100,500,230,460);
		ctx.quadraticCurveTo(260,440,250,475);
		ctx.stroke();
		ctx.fill();
		
		ctx.beginPath();
		//Planet at equ is (200,400)
		//(600,0) at north pole
		//(600,800) at south pole
		ctx.moveTo(650,740);
		ctx.bezierCurveTo(650,770,600,765,580,740);
		ctx.quadraticCurveTo(555,700,510,725);
		ctx.bezierCurveTo(495,735,465,700,480,690);
		ctx.quadraticCurveTo(495,660,480,640);
		ctx.bezierCurveTo(450,620,460,595,480,600);
		ctx.quadraticCurveTo(530,610,570,590);
		ctx.bezierCurveTo(620,550,700,700,700,750);
	/*	ctx.quadraticCurveTo();
		ctx.bezierCurveTo();
		ctx.quadraticCurveTo();*/
		ctx.stroke();
		ctx.fill();
	},
	LoadPlanet:function(old) {
		this.mods.food = old.mods.food;
		this.mods.wood = old.mods.wood;
		this.mods.metal = old.mods.metal;
		this.cosmetic.color1ndx = old.cosmetic.color1ndx;
		this.cosmetic.color1 = old.cosmetic.color1;
		this.cosmetic.color2ndx = old.cosmetic.color2ndx;
		this.cosmetic.color2 = old.cosmetic.color2;
		this.satellites = old.satellites;
		this.name = old.name;
		/*this.radius = old.radius;
		this.originX = old.originX;
		this.originY = old.originY;
		this.startAngle = old.startAngle;
		this.endAngle = old.endAngle;
		this.startCoord = old.startCoord;
		this.endCoord = old.endCoord;*/
		return this;
	}
}


pixel/js/Upgrades.js
function Upgrades() {
	this.upgrades = Array();
	this.upgrades[0] = {
		id: 0,
		name: "Pixel Display",
		desc: "Display your hard earned pixels",
		cost: 500,
		persist: false,
		prereq: -1,
		unlockFunction: Pixel.DisplayPixels
	};
	this.upgrades[1] = {
		id: 1,
		name: "Automatic Cursor",
		desc: "Automatically collects 1 pixels every 2 seconds for you",
		cost: 1000,
		persist: false,
		prereq: 0,
		unlockFunction: Pixel.AutoCursor
	};
	this.upgrades[2] = {
		id: 2,
		name: "Auto Cursor Speed",
		desc: "Cursors collect pixels faster",
		cost: 2500,
		costFunc: Pixel.AutoCursorSpeedCost,
		tracker: function() { return Pixel.State.autoCursorSpeedLvl; },
		persist: true,
		prereq: 1,
		unlockFunction: Pixel.AutoCursorSpeedUpgrade
	};
	this.upgrades[4] = {
		id: 4,
		name: "Cursor Size",
		desc: "Manually collect more pixels at once",
		cost: 5000,
		costFunc: Pixel.CursorSizeCost,
		tracker: function() { return Pixel.State.cursorSizeLvl; },
		persist: true,
		prereq: 1,
		unlockFunction: Pixel.CursorSizeUpgrade
	};
	this.upgrades[5] = {
		id: 5,
		name: "Basic Image Info",
		desc: "Shows the most basic image information",
		cost: 2000,
		persist: false,
		prereq: 1,
		unlockFunction: Pixel.BasicInfoUnlock
	};
	this.upgrades[6] = {
		id: 6,
		name: "Advanced Image Info",
		desc: "Shows more image information",
		cost: 8000,
		persist: false,
		prereq: 7,
		unlockFunction: Pixel.AdvancedInfoUnlock
	};
	this.upgrades[7] = {
		id: 7,
		name: "Basic Stats",
		desc: "Shows some basic stats",
		cost: 4000,
		persist: false,
		prereq: 5,
		unlockFunction: Pixel.BasicStatsUnlock
	};
	this.upgrades[8] = {
		id: 8,
		name: "Advanced Stats",
		desc: "Shows more statistics",
		cost: 12000,
		persist: false,
		prereq: 6,
		unlockFunction: Pixel.AdvancedStatsUnlock
	};
	this.upgrades[9] = {
		id: 9,
		name: "Get New Image",
		desc: "Gets a new image if you don't like the one you have<br />Only works about once every 30 seconds",
		cost: 10000,
		persist: true,
		tracker: false,
		prereq: 1,
		unlockFunction: Pixel.SkipImage
	};
	this.upgrades[10] = {
		id: 10,
		name: "Cursor Bomb",
		desc: "Drops a pixel collecting bomb when you click proportional to your cursor size",
		cost: 8000,
		persist: false,
		prereq: 1,
		unlockFunction: Pixel.CursorBombUnlock
	};
	this.upgrades[11] = {
		id: 11,
		name: "Bomb Size I",
		desc: "Makes the cursor bomb bigger",
		cost: 50000,
		persist: false,
		prereq: 10,
		unlockFunction: Pixel.CursorBombSizeUpgradeI
	};
	this.upgrades[12] = {
		id: 12,
		name: "Bomb Reload Speed",
		desc: "Makes the cursor bomb reload faster",
		cost: 10000,
		costFunc: Pixel.CursorBombSpeedCost,
		tracker: function() { return Pixel.State.cursorBombSpeedLvl; },
		persist: true,
		prereq: 10,
		unlockFunction: Pixel.CursorBombSpeedUpgrade
	};
	this.upgrades[13] = {
		id: 13,
		name: "Bomb Size II",
		desc: "Makes the cursor bomb bigger",
		cost: 60000,
		persist: false,
		prereq: 11,
		unlockFunction: Pixel.CursorBombSizeUpgradeII
	};
	this.upgrades[14] = {
		id: 14,
		name: "Bomb Size III",
		desc: "Makes the cursor bomb bigger",
		cost: 150000,
		persist: false,
		prereq: 13,
		unlockFunction: Pixel.CursorBombSizeUpgradeIII
	};
	this.upgrades[15] = {
		id: 15,
		name: "Bomb Size IV",
		desc: "Makes the cursor bomb bigger",
		cost: 600000,
		persist: false,
		prereq: 14,
		unlockFunction: Pixel.CursorBombSizeUpgradeIV
	};
	this.upgrades[16] = {
		id: 16,
		name: "Bomb Size V",
		desc: "Makes the cursor bomb bigger",
		cost: 2500000,
		persist: false,
		prereq: 15,
		unlockFunction: Pixel.CursorBombSizeUpgradeV
	};
	this.upgrades[17] = {
		id: 17,
		name: "Bomb Chain Chance",
		desc: "Increases chance of bombs triggering another bomb",
		cost: 10000,
		costFunc: Pixel.CursorBombChainCost,
		tracker: function() { return Pixel.State.cursorBombChainLvl; },
		persist: true,
		prereq: 10,
		unlockFunction: Pixel.CursorBombChainUpgrade
	};
	this.upgrades[18] = {
		id: 18,
		name: "Auto Finish Image",
		desc: "Unlocks a toggle to have the game automatically start the next image once complete",
		cost: 75000,
		persist: false,
		prereq: 1,
		unlockFunction: Pixel.AutoFinishImage
	};
	this.upgrades[19] = {
		id: 19,
		name: "Bomb Max Chain",
		desc: "Increases number of bombs that can be triggered in a chain. There are diminishing returns on successive bombings.",
		cost: 4000,
		costFunc: Pixel.CursorBombMaxChainCost,
		tracker: function() { return Pixel.State.cursorBombMaxChainLvl; },
		persist: true,
		prereq: 10,
		unlockFunction: Pixel.CursorBombMaxChainUpgrade
	};
	this.upgrades[20] = {
		id: 20,
		name: "Pixel Color",
		desc: "Unlocks a slider that allows you to change the pixel color",
		cost: 50000,
		persist: false,
		prereq: 1,
		unlockFunction: Pixel.PixelColorUnlock
	};
	this.upgrades[21] = {
		id: 21,
		name: "Pix Splitter",
		desc: "Splits pixels gathered manually, having them count for double currency",
		cost: 50000,
		persist: false,
		prereq: 1,
		unlockFunction: Pixel.PixelSplitManual1Unlock
	};
	this.upgrades[22] = {
		id: 22,
		name: "Pix Dicer",
		desc: "Further splits pixels gathered manually, they now count for quadruple currency",
		cost: 150000,
		persist: false,
		prereq: 21,
		unlockFunction: Pixel.PixelSplitManual2Unlock
	};
	this.upgrades[23] = {
		id: 23,
		name: "Auto Pix Splitter",
		desc: "Splits pixels gathered with the auto cursor, having them count for double currency",
		cost: 50000,
		persist: false,
		prereq: 1,
		unlockFunction: Pixel.PixelSplitAuto1Unlock
	};
	this.upgrades[24] = {
		id: 24,
		name: "Auto Pix Dicer",
		desc: "Further splits pixels gathered with the auto cursor, they now count for quadruple currency",
		cost: 150000,
		persist: false,
		prereq: 23,
		unlockFunction: Pixel.PixelSplitAuto2Unlock
	};
	this.upgrades[25] = {
		id: 25,
		name: "Bomb Pix Splitter",
		desc: "Splits pixels gathered with the bomb, having them count for double currency",
		cost: 50000,
		persist: false,
		prereq: 10,
		unlockFunction: Pixel.PixelSplitBomb1Unlock
	};
	this.upgrades[26] = {
		id: 26,
		name: "Bomb Pix Dicer",
		desc: "Further splits pixels gathered with the bomb, they now count for quadruple currency",
		cost: 150000,
		persist: false,
		prereq: 25,
		unlockFunction: Pixel.PixelSplitBomb2Unlock
	};
	this.upgrades[27] = {
		id: 27,
		name: "Party Pixel Spawn",
		desc: "Cuts the time between party pixels in half",
		cost: 250000,
		persist: false,
		prereq: 1,
		unlockFunction: Pixel.PartyPixelSpawn
	};
	this.upgrades[28] = {
		id: 28,
		name: "Party Pixel Pop",
		desc: "Gives bonus pixels when you pop a party pixel",
		cost: 10000,
		costFunc: Pixel.PartyPixelPopCost,
		tracker: function() { return Pixel.State.partyPixelPopLvl; },
		persist: true,
		prereq: 1,
		unlockFunction: Pixel.PartyPixelPop
	};
    this.upgrades[29] = {
        id: 29,
        name: "Party Pixel Party",
        desc: "Pixel Parties last twice as long",
        cost: 1250000,
        persist: false,
        prereq: 27,
        unlockFunction: Pixel.PartyPixelDuration
    };
    this.upgrades[30] = {
        id: 30,
        name: "NSFW Toggle",
        desc: "Lets you toggle the ability to see NSFW images, not to see only NSFW images",
        cost: 1000000,
        persist: false,
        prereq: 31,
        unlockFunction: Pixel.NsfwUnlock
    };
    this.upgrades[31] = {
        id: 31,
        name: "Search Filter",
        desc: "Allows you to specifiy a subreddit to pull images from or a search term to filter",
        cost: 250000,
        persist: false,
        prereq: 8,
        unlockFunction: Pixel.SearchTermUnlock
    };

	this.upgradeList = Array(0,7,8,5,6,31,30,9,4,1,2,10,12,11,13,14,15,16,17,19,18,20,21,22,23,24,25,26,28,27,29);
	this.owned = Array();
}

Upgrades.prototype = {
	constructor: Upgrades,
	LoadUpgrades:function(old) {
		this.owned = old.owned;
		return this;
	},
	Check:function(ndx) {
		return !($.inArray(ndx, this.owned) == -1);
	}
};


pixel/js/Achievements.js
function Achievements() {
	this.chievos = Array();
	this.chievos[0] = {
		name: "Depixelator I",
		hidden: false,
		beforeDesc: "Uncover 1 pixel",
		desc: "Uncovered 1 pixel",
		image: "depixel1.png"
	};
	this.chievos[1] = {
		name: "Depixelator II",
		hidden: false,
		beforeDesc: "Uncover 500 pixels",
		desc: "Uncovered 500 pixels, now you can buy things!",
		image: "depixel2.png"
	};
	this.chievos[2] = {
		name: "Depixelator III",
		hidden: false,
		beforeDesc: "Uncover 1000 pixels",
		desc: "Uncovered 1000 pixels",
		image: "depixel3.png"
	};
	this.chievos[3] = {
		name: "Depixelator IV",
		hidden: false,
		beforeDesc: "Uncover 2500 pixels",
		desc: "Uncovered 2500 pixels",
		image: "depixel4.png"
	};
	this.chievos[4] = {
		name: "Depixelator V",
		hidden: false,
		beforeDesc: "Uncover 5000 pixels",
		desc: "Uncovered 5000 pixels",
		image: "depixel5.png"
	};
	this.chievos[5] = {
		name: "Depixelator VI",
		hidden: false,
		beforeDesc: "Uncover 10000 pixels",
		desc: "Uncovered 10000 pixels, now we're getting somewhere",
		image: "depixel6.png"
	};
	this.chievos[6] = {
		name: "Depixelator VII",
		hidden: false,
		beforeDesc: "Uncover 25000 pixels",
		desc: "Uncovered 25000 pixels",
		image: "depixel7.png"
	};
	this.chievos[7] = {
		name: "Depixelator VIII",
		hidden: false,
		beforeDesc: "Uncover 50000 pixels",
		desc: "Uncovered 50000 pixels",
		image: "depixel8.png"
	};
	this.chievos[8] = {
		name: "Depixelator IX",
		hidden: false,
		beforeDesc: "Uncover 100k pixels",
		desc: "Uncovered 100k pixels",
		image: "depixel9.png"
	};
	this.chievos[9] = {
		name: "Depixelator X",
		hidden: false,
		beforeDesc: "Uncover 500k pixels",
		desc: "Uncovered 500k pixels",
		image: "depixel10.png"
	};
	this.chievos[10] = {
		name: "Mega-Depixelator I",
		hidden: false,
		beforeDesc: "Uncover 1M pixels",
		desc: "Uncovered 1M pixels",
		image: "megadepixel1.png"
	};
	this.chievos[11] = {
		name: "Mega-Depixelator II",
		hidden: false,
		beforeDesc: "Uncover 5M pixels",
		desc: "Uncovered 5M pixels",
		image: "megadepixel2.png"
	};
	this.chievos[12] = {
		name: "Mega-Depixelator III",
		hidden: false,
		beforeDesc: "Uncover 10M pixels",
		desc: "Uncovered 10M pixels",
		image: "megadepixel3.png"
	};
	this.chievos[13] = {
		name: "Mega-Depixelator IV",
		hidden: false,
		beforeDesc: "Uncover 100M pixels",
		desc: "Uncovered 100M pixels",
		image: "megadepixel4.png"
	};
	this.chievos[14] = {
		name: "Mega-Depixelator V",
		hidden: false,
		beforeDesc: "Uncover 1B pixels",
		desc: "Uncovered 1B pixels",
		image: "megadepixel5.png"
	};
	this.chievos[15] = {
		name: "Mega-Depixelator VI",
		hidden: false,
		beforeDesc: "Uncover 10B pixels",
		desc: "Uncovered 10B pixels",
		image: "megadepixel6.png"
	};
	this.chievos[16] = {
		name: "Mega-Depixelator VII",
		hidden: false,
		beforeDesc: "Uncover 100B pixels",
		desc: "Uncovered 100B pixels",
		image: "megadepixel7.png"
	};
	this.chievos[17] = {
		name: "Mega-Depixelator VIII",
		hidden: false,
		beforeDesc: "Uncover 1T pixels",
		desc: "Uncovered 1T pixels",
		image: "megadepixel8.png"
	};
	this.chievos[18] = {
		name: "Mega-Depixelator IX",
		hidden: false,
		beforeDesc: "Uncover 10T pixels",
		desc: "Uncovered 10T pixels",
		image: "megadepixel9.png"
	};
	this.chievos[19] = {
		name: "Mega-Depixelator X",
		hidden: false,
		beforeDesc: "Uncover 100T pixels",
		desc: "Uncovered 100T pixels",
		image: "megadepixel10.png"
	};
	this.chievos[20] = {
		name: "Manual Picker I",
		hidden: false,
		beforeDesc: "Manually uncover 1k pixels",
		desc: "Manually uncovered 1k pixels",
		image: "manual1.png"
	};
	this.chievos[21] = {
		name: "Manual Picker II",
		hidden: false,
		beforeDesc: "Manually uncover 1M pixels",
		desc: "Manually uncovered 1M pixels",
		image: "manual2.png"
	};
	this.chievos[22] = {
		name: "Manual Picker III",
		hidden: false,
		beforeDesc: "Manually uncover 1B pixels",
		desc: "Manually uncovered 1B pixels",
		image: "manual3.png"
	};
	this.chievos[23] = {
		name: "Manual Picker IV",
		hidden: false,
		beforeDesc: "Manually uncover 1T pixels",
		desc: "Manually uncovered 1T pixels",
		image: "manual4.png"
	};
	this.chievos[24] = {
		name: "Manual Picker V",
		hidden: false,
		beforeDesc: "Manually uncover 1Qu pixels",
		desc: "Manually uncovered 1Qu pixels",
		image: "manual5.png"
	};
	this.chievos[30] = {
		name: "Auto Picker I",
		hidden: false,
		beforeDesc: "Automatically uncover 1k pixels",
		desc: "Automatically uncovered 1k pixels",
		image: "auto1.png"
	};
	this.chievos[31] = {
		name: "Auto Picker II",
		hidden: false,
		beforeDesc: "Automatically uncover 1M pixels",
		desc: "Automatically uncovered 1M pixels",
		image: "auto2.png"
	};
	this.chievos[32] = {
		name: "Auto Picker III",
		hidden: false,
		beforeDesc: "Automatically uncover 1B pixels",
		desc: "Automatically uncovered 1B pixels",
		image: "auto3.png"
	};
	this.chievos[33] = {
		name: "Auto Picker IV",
		hidden: false,
		beforeDesc: "Automatically uncover 1T pixels",
		desc: "Automatically uncovered 1T pixels",
		image: "auto4.png"
	};
	this.chievos[34] = {
		name: "Auto Picker V",
		hidden: false,
		beforeDesc: "Automatically uncover 1Qu pixels",
		desc: "Automatically uncovered 1Qu pixels",
		image: "auto5.png"
	};
	this.chievos[40] = {
		name: "Bomber I",
		hidden: false,
		beforeDesc: "Uncover 1k pixels with bombs",
		desc: "Uncovered 1k pixels with bombs",
		image: "bomb1.png"
	};
	this.chievos[41] = {
		name: "Bomber II",
		hidden: false,
		beforeDesc: "Uncover 1M pixels with bombs",
		desc: "Uncovered 1M pixels with bombs",
		image: "bomb2.png"
	};
	this.chievos[42] = {
		name: "Bomber III",
		hidden: false,
		beforeDesc: "Uncover 1B pixels with bombs",
		desc: "Uncovered 1B pixels with bombs",
		image: "bomb3.png"
	};
	this.chievos[43] = {
		name: "Bomber IV",
		hidden: false,
		beforeDesc: "Uncover 1T pixels with bombs",
		desc: "Uncovered 1T pixels with bombs",
		image: "bomb4.png"
	};
	this.chievos[44] = {
		name: "Bomber V",
		hidden: false,
		beforeDesc: "Uncover 1Qu pixels with bombs",
		desc: "Uncovered 1Qu pixels with bombs",
		image: "bomb5.png"
	};
	this.chievos[45] = {
		name: "Speedy",
		hidden: false,
		beforeDesc: "Finish a picture in under 2700 seconds",
		desc: "Finished a picture in under 2700 seconds",
		image: "speedy1.png"
	};
	this.chievos[46] = {
		name: "Speedier",
		hidden: false,
		beforeDesc: "Finish a picture in under 1800 seconds",
		desc: "Finished a picture in under 1800 seconds",
		image: "speedy2.png"
	};
	this.chievos[47] = {
		name: "Speediest",
		hidden: false,
		beforeDesc: "Finish a picture in under 1200 seconds",
		desc: "Finished a picture in under 1200 seconds",
		image: "speedy3.png"
	};
	this.chievos[50] = {
		name: "Most Speediest",
		hidden: true,
		beforeDesc: "Finish a picture in under 600 seconds",
		desc: "Finished a picture in under 600 seconds",
		image: "speedy4.png"
	};
	this.chievos[51] = {
		name: "Chain Master",
		hidden: true,
		beforeDesc: "Get a bomb chain of more than 8",
		desc: "Got a bomb chain of more than 8",
		image: "chain.png"
	};
	this.chievos[52] = {
		name: "Information Addict",
		hidden: true,
		beforeDesc: "Unlock all the image information upgrades",
		desc: "Unlocked all the image information upgrades",
		image: "infoaddict.png"
	};
	this.chievos[53] = {
		name: "Statistics Addict",
		hidden: true,
		beforeDesc: "Unlock all the statistic upgrades",
		desc: "Unlocked all the statistic upgrades",
		image: "stataddict.png"
	};
	this.chievos[54] = {
		name: "Completionist",
		hidden: true,
		beforeDesc: "Uncover every pixel in an image before moving on (Auto only works to 99.9%)",
		desc: "Uncovered every pixel in an image",
		image: "completionist.png"
	};
	this.chievos[55] = {
		name: "Bored",
		hidden: true,
		beforeDesc: "Finish an image without manually collecting any pixels",
		desc: "Finished an image without manually collecting any pixels",
		image: "bored.png"
	};
	this.chievos[56] = {
		name: "Got Those",
		hidden: true,
		beforeDesc: "Try to uncover 10000 pixels that have already been uncovered",
		desc: "Uncovered 10000 pixels that have already been uncovered",
		image: "gotthose.png"
	};
	this.chievos[57] = {
		name: "True Speedster",
		hidden: true,
		beforeDesc: "Finish a picture in under 60 seconds",
		desc: "Finished a picture in under 60 seconds",
		image: "speedy5.png"
	};
	this.chievos[60] = {
		name: "Picturesque I",
		hidden: false,
		beforeDesc: "Finish 1 picture",
		desc: "Finished 1 picture",
		image: "picture1.png"
	};
	this.chievos[61] = {
		name: "Picturesque II",
		hidden: false,
		beforeDesc: "Finish 5 pictures",
		desc: "Finished 5 pictures",
		image: "picture2.png"
	};
	this.chievos[62] = {
		name: "Picturesque III",
		hidden: false,
		beforeDesc: "Finish 25 pictures",
		desc: "Finished 25 pictures",
		image: "picture3.png"
	};
	this.chievos[63] = {
		name: "Picturesque IV",
		hidden: false,
		beforeDesc: "Finish 100 pictures",
		desc: "Finished 100 pictures",
		image: "picture4.png"
	};
	this.chievos[64] = {
		name: "Picturesque V",
		hidden: false,
		beforeDesc: "Finish 1000 pictures",
		desc: "Finished 1000 pictures",
		image: "picture5.png"
	};
	this.chievos[70] = {
		name: "Uncultured Swine I",
		hidden: false,
		beforeDesc: "Skip 1 picture",
		desc: "Skipped 1 picture",
		image: "skip1.png"
	};
	this.chievos[71] = {
		name: "Uncultured Swine II",
		hidden: false,
		beforeDesc: "Skip 5 pictures",
		desc: "Skipped 5 pictures",
		image: "skip2.png"
	};
	this.chievos[72] = {
		name: "Uncultured Swine III",
		hidden: false,
		beforeDesc: "Skip 25 pictures",
		desc: "Skipped 25 pictures",
		image: "skip3.png"
	};
	this.chievos[73] = {
		name: "Uncultured Swine IV",
		hidden: false,
		beforeDesc: "Skip 50 pictures",
		desc: "Skipped 50 pictures",
		image: "skip4.png"
	};
	this.chievos[74] = {
		name: "Uncultured Swine V",
		hidden: false,
		beforeDesc: "Skip 100 pictures",
		desc: "Skipped 100 pictures",
		image: "skip5.png"
	};
	this.chievos[80] = {
		name: "Getting Started I",
		hidden: false,
		beforeDesc: "Uncover 1% of your first image",
		desc: "Uncovered 1% of your first image",
		image: "start1.png"
	};
	this.chievos[81] = {
		name: "Getting Started II",
		hidden: false,
		beforeDesc: "Uncover 10% of your first image",
		desc: "Uncovered 10% of your first image",
		image: "start2.png"
	};
	this.chievos[82] = {
		name: "Getting Started III",
		hidden: false,
		beforeDesc: "Uncover 25% of your first image",
		desc: "Uncovered 25% of your first image",
		image: "start3.png"
	};
	this.chievos[83] = {
		name: "Getting Started IV",
		hidden: false,
		beforeDesc: "Uncover 50% of your first image",
		desc: "Uncovered 50% of your first image",
		image: "start4.png"
	};
	this.chievos[84] = {
		name: "Getting Started V",
		hidden: false,
		beforeDesc: "Uncover 75% of your first image",
		desc: "Uncovered 75% of your first image",
		image: "start5.png"
	};
	this.chievos[85] = {
		name: "Getting Started VI",
		hidden: false,
		beforeDesc: "Uncover 99% of your first image",
		desc: "Uncovered 99% of your first image",
		image: "start6.png"
	};
	this.chievos[90] = {
		name: "Partier I",
		hidden: false,
		beforeDesc: "Have a pixel party!",
		desc: "You had your first pixel party!",
		image: "party1.png"
	};
	this.chievos[91] = {
		name: "Partier II",
		hidden: false,
		beforeDesc: "Have 10 pixel parties",
		desc: "You've had 10 pixel parties",
		image: "party2.png"
	};
	this.chievos[92] = {
		name: "Partier III",
		hidden: false,
		beforeDesc: "Have 50 pixel parties",
		desc: "You've had 50 pixel parties",
		image: "party3.png"
	};
	this.chievos[93] = {
		name: "Partier IV",
		hidden: false,
		beforeDesc: "Have 100 pixel parties",
		desc: "You've had 100 pixel parties",
		image: "party4.png"
	};
	this.chievos[94] = {
		name: "Partier V",
		hidden: false,
		beforeDesc: "Have 1000 pixel parties",
		desc: "You've had 1000 pixel parties",
		image: "party5.png"
	};
	this.chievos[95] = {
		name: "Party Pooper",
		hidden: true,
		beforeDesc: "Miss 1000 pixel parties",
		desc: "You've missed 1000 pixel parties :-(",
		image: "partypooper.png"
	};
	this.chievos[100] = {
		name: "Bomberman I",
		hidden: false,
		beforeDesc: "Launch 1 bomb",
		desc: "Launched your first bomb!",
		image: "bomber1.png"
	};
	this.chievos[101] = {
		name: "Bomberman II",
		hidden: false,
		beforeDesc: "Explode 50 bombs",
		desc: "50 bombs have exploded",
		image: "bomber2.png"
	};
	this.chievos[102] = {
		name: "Bomberman III",
		hidden: false,
		beforeDesc: "Explode 250 bombs",
		desc: "250 bombs have exploded",
		image: "bomber3.png"
	};
	this.chievos[103] = {
		name: "Bomberman IV",
		hidden: false,
		beforeDesc: "Explode 500 bombs",
		desc: "500 bombs have exploded",
		image: "bomber4.png"
	};
	this.chievos[104] = {
		name: "Bomberman V",
		hidden: false,
		beforeDesc: "Explode 1000 bombs",
		desc: "1000 bombs have exploded",
		image: "bomber5.png"
	};
	this.chievos[105] = {
		name: "Pacifist",
		hidden: true,
		beforeDesc: "Complete an image without launching a bomb",
		desc: "Completed an image without using any bombs",
		image: "pacifist1.png"
	};
	this.chievoList = Array(80,81,82,83,84,85,0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,30,31,32,33,34,40,41,
		42,43,44,45,46,47,60,61,62,63,64,70,71,72,73,74,50,57,51,52,53,54,55,56,90,91,92,93,94,95,100,101,102,103,104,105);
	this.achieved = Array();
	this.achievedOn = Array();
}

Achievements.prototype = {
	constructor: Achievements,
	LoadAchievements:function(old) {
		this.achieved = old.achieved;
		if(old.achievedOn !== undefined){
		    this.achievedOn = old.achievedOn;
		}
		return this;
	},
	ToastAchievement:function(ndx) {
		$('#toaster').append("<div class='toastDiv' id='toast"+ndx+"'><div class='toastHeader'>Achievement Unlocked!</div><div class='toastChievo'>"+
			this.chievos[ndx].name+"</div><span class='toastDesc'>"+this.chievos[ndx].desc+"</span></div>");
		$('#toast'+ndx).delay(2000).fadeOut(6000);
	},
	UnlockAchievement:function(ndx, link) {
		this.achieved.push(ndx);
		this.achievedOn[this.achieved.length-1] = link;
		this.ToastAchievement(ndx);
	}
}
